File: backend/src/main/java/zw/co/hisolutions/core/service/ProductTypeService.java
package zw.co.hisolutions.core.service;
 
import zw.co.hisolutions.core.entity.ProductType;
import zw.co.hisolutions.common.service.GenericService;

public interface ProductTypeService extends GenericService<ProductType, Long>{
    public ProductType findByName(String name);   
}

File: frontend/src/app/admin/services/rest/product-type.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http'; 

import {RestDataService} from 'shared/services/rest-data.service';

@Injectable()
export class ProductTypeService extends RestDataService {

    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/producttypes'; 
        super(httpClient, resourcePath);
    }

}
